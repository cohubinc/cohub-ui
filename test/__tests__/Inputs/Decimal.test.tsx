import React from "react";
import { mount } from "enzyme";

import { Inputs } from "dist";

describe("Decimal", () => {
  it("renders Decimal without crashing", () => {
    const wrapper = mount(
      <Inputs.Decimal input={{ value: 0, onChange: _value => undefined }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
