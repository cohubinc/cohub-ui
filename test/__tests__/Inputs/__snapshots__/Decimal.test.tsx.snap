// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Decimal renders Decimal without crashing 1`] = `
<DecimalInput
  input={
    Object {
      "onChange": [Function],
      "value": 0,
    }
  }
>
  <div
    className="styles-module_input__28syv"
  >
    <FloatingLabelWrapper
      error={false}
      onChange={[Function]}
      value={0}
    >
      <div
        className="FloatingLabelWrapper GenericInput "
        style={
          Object {
            "color": "hsl(0, 2%, 12%)",
            "cursor": "text",
            "position": "relative",
          }
        }
      >
        <div
          className="noLabelInputWrapper  bd-radius"
        >
          <NumberFormat
            allowEmptyFormatting={false}
            allowLeadingZeros={false}
            allowNegative={true}
            aria-invalid={false}
            decimalScale={2}
            decimalSeparator="."
            displayType="input"
            fixedDecimalScale={false}
            getInputRef={[Function]}
            isAllowed={[Function]}
            isNumericString={false}
            onBlur={[Function]}
            onChange={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseUp={[Function]}
            onValueChange={[Function]}
            prefix=""
            style={
              Object {
                "color": "hsl(0, 2%, 12%)",
                "cursor": "text",
                "textAlign": undefined,
              }
            }
            suffix=""
            thousandSeparator={true}
            thousandsGroupStyle="thousand"
            type="text"
            value={0}
          >
            <input
              aria-invalid={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "color": "hsl(0, 2%, 12%)",
                  "cursor": "text",
                  "textAlign": undefined,
                }
              }
              type="text"
              value="0"
            />
          </NumberFormat>
          <span
            className="bar "
          />
        </div>
      </div>
    </FloatingLabelWrapper>
  </div>
</DecimalInput>
`;
